### PR TITLE
only update powerstate if sure it is the latest

### PR DESCRIPTION
**Pull request by DaanHoogland**
## Description

On update powerstate check if we have the latest knowledge before doing so.
When report is missing don't try to stop the VM on the host that doesn't know the VM.

Fixes: #3741 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

This is purely based on log- and code analysis. intensiove review is requested. Also advise on environmental consequences is welcomed.
HA may be impacted do I suspect not in a negative way.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by DaanHoogland** [comment]
any vmware users or HA users are invited to have a look at this.

**Comment by DaanHoogland** [comment]
will be backporting to 4.13

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by andrijapanicsb** [comment]
This needs testing with VMware specifically

**Comment by andrijapanicsb** [comment]
@blueorangutan test centos7 vmware65u2

**Comment by andrijapanicsb** [comment]
@blueorangutan  test centos7 vmware-65u2

**Comment by andrijapanicsb** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by DaanHoogland** [comment]
revert snapshot error does not seem related but doing a re-package/re-test to be sure.
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by andrijapanicsb** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-65u2

File: engine/orchestration/src/main/java/com/cloud/vm/VirtualMachinePowerStateSyncImpl.java [hunk 1]
```diff
@@ -130,23 +136,34 @@ private void processReport(long hostId, Map<Long, VirtualMachine.PowerState> tra
                     }
                 }
 
-                if (s_logger.isDebugEnabled())
-                    s_logger.debug("Detected missing VM. host: " + hostId + ", vm id: " + instance.getId() +
-                            ", power state: PowerReportMissing, last state update: " + vmStateUpdateTime.getTime());
+                if (s_logger.isInfoEnabled()) {
+                    s_logger.info(
+                            String.format("Detected missing VM. host: %l, vm id: %l(%s), power state: %s, last state update: %l"
```
Comment (nvazquez) [inline engine/orchestration/src/main/java/com/cloud/vm/VirtualMachinePowerStateSyncImpl.java]: Hi @DaanHoogland, I'm getting this error in latest master:

````
INFO  [c.c.a.m.AgentManagerImpl] (AgentTaskPool-98:ctx-ad691272) (logid:1a451217) The agent from host 1 state determined is Up
INFO  [c.c.a.m.AgentManagerImpl] (AgentTaskPool-98:ctx-ad691272) (logid:1a451217) Agent is determined to be up and running
WARN  [c.c.a.m.AgentManagerImpl] (AgentManager-Handler-8:null) (logid:) Caught: 
java.util.UnknownFormatConversionException: Conversion = 'l'
	at java.util.Formatter$FormatSpecifier.conversion(Formatter.java:2691)
	at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2720)
	at java.util.Formatter.parse(Formatter.java:2560)
	at java.util.Formatter.format(Formatter.java:2501)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
	at com.cloud.vm.VirtualMachinePowerStateSyncImpl.processReport(VirtualMachinePowerStateSyncImpl.java:141)
	at com.cloud.vm.VirtualMachinePowerStateSyncImpl.processHostVmStatePingReport(VirtualMachinePowerStateSyncImpl.java:66)
	at com.cloud.vm.VirtualMachineManagerImpl.processCommands(VirtualMachineManagerImpl.java:3245)
	at com.cloud.agent.manager.AgentManagerImpl.handleCommands(AgentManagerImpl.java:309)
	at com.cloud.agent.manager.AgentManagerImpl$AgentHandler.processRequest(AgentManagerImpl.java:1287)
	at com.cloud.agent.manager.AgentManagerImpl$AgentHandler.doTask(AgentManagerImpl.java:1370)
	at com.cloud.agent.manager.ClusteredAgentManagerImpl$ClusteredAgentHandler.doTask(ClusteredAgentManagerImpl.java:702)
	at com.cloud.utils.nio.Task.call(Task.java:83)
	at com.cloud.utils.nio.Task.call(Task.java:29)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
````

Comment (DaanHoogland) [reply engine/orchestration/src/main/java/com/cloud/vm/VirtualMachinePowerStateSyncImpl.java]: yes, i know, fixed in #3806

**Pull request merged**